test/engine/test_execute.py

```diff
@@ -707,6 +707,36 @@ def test_works_after_dispose_testing_engine(self):
                 eq_(conn.scalar(select(1)), 1)
             eng.dispose()
 
+    @testing.requires.qmark_paramstyle
+    def test_scalar(self, connection):
+        conn = connection
+        conn.exec_driver_sql(
+            "insert into users (user_id, user_name) " "values (?, ?)",
```

Comment:
**mike bayer** (zzzeek) wrote:

the "?" is actually not named paramstyle.    For these tests we dont have to deal with any of that explicitly.     These two tests can be written as follows:


    def test_scalar(self, connection):
        conn = connection
        users = self.tables.users
        conn.execute(
            users.insert(),
            [
                {"user_id": 1, "user_name": "sandy"},
                {"user_id": 2, "user_name": "spongebob"},
            ],
        )
        res = conn.scalar(
            select(users.c.user_name).order_by(users.c.user_id)
        )
        eq_(res, "sandy")

    def test_scalars(self, connection):
        conn = connection
        users = self.tables.users
        conn.execute(
            users.insert(),
            [
                {"user_id": 1, "user_name": "sandy"},
                {"user_id": 2, "user_name": "spongebob"},
            ],
        )
        res = conn.scalars(
            select(users.c.user_name).order_by(users.c.user_id)
        )
        eq_(res.all(), ["sandy", "spongebob"])

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3065